packages/grid/_modules_/grid/components/panel/filterPanel/GridFilterInputValue.tsx

```diff
@@ -47,7 +48,11 @@ function GridFilterInputValue(props: GridTypeFilterInputValueProps & TextFieldPr
           ),
         }
       : {};
-
+  React.useEffect(() => {
```

Comment:
I think we should use the same behavior / naming convention as in the grid cells

`autoFocus` => `hasFocus` (to keep the distinction `autoFocus` = set focus on mount and `hasFocus` = set focus when `true`)
`React.useEffect` => `useEnhancedEffet` (to set the focus sooner)